bootstrap/result.go

```diff
@@ -65,6 +65,9 @@ func (sr *shardResult) AddSeries(id string, rawSeries m3db.DatabaseSeriesBlocks)
 
 // AddResult adds a shard result.
 func (sr *shardResult) AddResult(other m3db.ShardResult) {
+	if other == nil {
+		return
+	}
```

Comment:
Hm, where are we calling `AddResult(nil)` from?  This should probably ok, however calling `append(nil)` being a no-op in most places doesn't seem like a no-op, e.g. `[1,2,3].append(nil)` you would expect to be `[1,2,3,nil]`.  However I guess this more like an `appendRange()` method?  `var range []*int; [1,2,3].appendRange(range)` is likely expected to be `[1,2,3]` like the behavior you're introducing here.